[Amitha} iP

amitha2210 opened the pull request:
### ColbyPro

> “If it’s on your mind, get it out of your mind. Write it down…Then take what’s on your mind and get things done.” - David Allen

ColbyPro frees your mind of having to remember things you need to do.   It's,

- text-based
- easy to learn
- SPEEDY to use

All you need to do is,

1. download it from [_here_]([here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html))
2. double-click it
3. add your tasks
4. let it manage your tasks for you! 😉

And it is **FREE**!

Features:

- [ ] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
______________________________________________________________________________________________________
If you are a Java programmer, you can use it to practice Java too. Here's the main method:

`public class Main {
     public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}`

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

amitha2210 committed:
Change name and say hi and bye

amitha2210 committed:
Echoes user input and says bye

amitha2210 committed:
Stores task in array and returns list

amitha2210 committed:
Mark and unmark

amitha2210 committed:
Todo, Event and deadline

amitha2210 committed:
Testing

amitha2210 committed:
Testing edited

amitha2210 committed:
Exceptions minimal

amitha2210 committed:
Delete

amitha2210 committed:
Add task items to a file on device

amitha2210 committed:
Save date of deadline the user has input

amitha2210 committed:
Add more OOP using more classes

amitha2210 committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

amitha2210 committed:
Add all files to package named colby

amitha2210 committed:
Add tests for methods from TaskList and Parser classes

amitha2210 committed:
Add one test to ParserTest file

amitha2210 committed:
Package as JAR file

amitha2210 committed:
Add JavaDoc comments

amitha2210 committed:
Comply with coding standard for line length

amitha2210 committed:
Comply with coding standard for import statements

amitha2210 committed:
Comply with coding standard by separating import statements

amitha2210 committed:
Merge branch 'branch-A-JavaDoc' into branch-A-CodingStandard

* branch-A-JavaDoc:
  Add JavaDoc comments

# Conflicts:
#	src/main/java/colby/Deadline.java

amitha2210 committed:
Merge commit '2d1cab1aaf7bac00bb82782f8e90b8a1a5a2ed4d'

* commit '2d1cab1aaf7bac00bb82782f8e90b8a1a5a2ed4d':
  Comply with coding standard for import statements
  Comply with coding standard for line length

# Conflicts:
#	src/main/java/colby/Storage.java

amitha2210 committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Add JavaDoc comments

amitha2210 committed:
Add find based on keyword feature

amitha2210 committed:
Add JavaDoc for new function for find feature

kvernnn commented on src/main/java/colby/Task.java:
```diff
@@ -0,0 +1,73 @@
+            return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    public void markAsDone() {
```
Could a shorter name for these methods be better? Such as 'mark' and 'unmark'

kvernnn commented on src/main/java/colby/TaskList.java:
```diff
@@ -0,0 +1,76 @@
+     * Adds a task to the list
+     * @param task task that will be added to the list
+     */
+    public void addTask(Task task) {
```
I really like how the methods follows the naming conventions, starting with a verb

kvernnn commented on src/main/java/colby/Event.java:
```diff
@@ -0,0 +1,27 @@
+ * start and end refer to the start time/date and end time/date of the event
+ */
+public class Event extends Task {
+    private final String start;
+    private final String end;
```
I really like the naming conventions for your variables, they are short yet clearly indicates what they represent

kvernnn commented on src/main/java/colby/Colby.java:
```diff
@@ -0,0 +1,76 @@
+        scanner.close();
+    }
+    public static void main(String[] args) throws IOException {
+        new Colby("Data.txt").run();
```
Would having a private static final FILE_NAME or something similar be more appropriate to replace "Data.txt", since it is always being used in the main method?

kvernnn reviewed (commented):
LGTM, just some nits to be reviewed

amitha2210 committed:
Add checkstyle files and run it

amitha2210 committed:
Edit build.gradle file for checkstyle

amitha2210 committed:
Merge branch 'branch-A-checkstyle'

* branch-A-checkstyle:
  Edit build.gradle file for checkstyle
  Add checkstyle files and run it
  Add JavaDoc for new function for find feature
  Add find based on keyword feature

# Conflicts:
#	src/main/java/colby/Ui.java

amitha2210 committed:
Add files from JavaFx tutorial for a guide

amitha2210 committed:
Adjust JavaFx relevant files and methods in UI class

amitha2210 committed:
Edit files such that todo, deadline and event commands can work

amitha2210 committed:
When command list is entered, only tasks entered in that respective run are returned, not all those stored in Data file

The feature of being able to store the tasks data and refer to them again is not being used, hence defeats the purpose of the chatbot being able to list all the tasks.

New method in Storage class that returns the String form of all tasks stored in Data file.

A String is returned instead of the usual printing of content directly, such that the String can be used to work with the JavaFx files as well.

amitha2210 committed:
No assertions in place

Some assumptions made are not checked, hence might cause a bug in the programme

Add assertions so check size of TaskList tasks

Helps to ensure the TaskList tasks' size is not invalid when tasks are added or deleted

amitha2210 committed:
Merge pull request #1 from amitha2210/branch-A-Assertions

No assertions in place

amitha2210 committed:
Remove unused method

amitha2210 committed:
Add function to note down duration required for ToDo task if included in input

amitha2210 committed:
Add function for fixed duration task for Todo task

amitha2210 committed:
Edit the string saved and returned for ToDo tasks to be in the correct format

amitha2210 committed:
Merge branch 'branch-A-CodeQuality'

* branch-A-CodeQuality:
  Remove unused method

amitha2210 committed:
Merge branch 'branch-fixedDurationTasks'

* branch-fixedDurationTasks:
  Edit the string saved and returned for ToDo tasks to be in the correct format
  Add function for fixed duration task for Todo task
  Add function to note down duration required for ToDo task if included in input

amitha2210 committed:
Add ReadMe page details

amitha2210 committed:
Change initial taskList to load existing tasks

amitha2210 committed:
Edit buildList method to include adding all details of relevant tasks

amitha2210 committed:
Fix bug for string saved to data file when todo command is entered

amitha2210 committed:
Fix how the status brackets are printed for the tasks

amitha2210 committed:
Add delete command to README file

amitha2210 committed:
Add find command to README file

amitha2210 committed:
Edit JavaFx and CSS components

amitha2210 committed:
Remove unused code and add some JavaDoc comments

amitha2210 committed:
Add more exception handling for date/time formatting
